source-map.bs

```diff
@@ -256,14 +364,166 @@ with four fields represent mapped code where a corresponding name does not exist
+                1. Continue with the next iteration.
+            <!-- TODO: If there is a relativeSourceIndex but not a relativeOriginalLine,
+                 should |sourceIndex| still be increased? -->
+            1. Increase |sourceIndex| by |relativeSourceIndex|.
+            1. Increase |originalLine| by |relativeOriginalLine|.
+            1. Increase |originalColumn| by |relativeOriginalPosition|.
+            1. If any of |sourceIndex|, |originalLine|, or |originalColumn| are less than 0, or if
+                |sourceIndex| is greater than or equal to |sources|'s [=list/size=], [=optionally
+                report an error=].
+            1. Else,
+                1. Set |decodedMapping|'s [=decoded mapping/originalSource=] to
+                    |sources|[|sourceIndex|].
+                1. Set |decodedMapping|'s [=decoded mapping/originalLine=] to |originalLine|.
+                1. Set |decodedMapping|'s [=decoded mapping/originalColumn=] to |originalColumn|.
+            1. [=Decode a base64 VLQ=] from |segment| given |position| and let |relativeNameIndex|
+                be the result.
+            1. If |relativeNameIndex| is not null, then:
+                1. Increase |nameIndex| by |relativeNameIndex|.
+                1. If |nameIndex| is negative or greater than |names|'s [=list/size=], [=optionally
+                    report an error=].
+                1. Else, set |decodedMapping|'s [=decoded mapping/name=] to |names|[|nameIndex|].
+            1. If |position| does not point to the end of |segment|, [=optionally report an
+                error=].
+    1. Increase |generatedLine| by 1.
+1. Return |decodedMappings|.
+
+To <dfn>decode a base64 VLQ</dfn> from a [=string=] |segment| given a [=position variable=]
+position, run the following stepsis the number :
+1. Let |first| be a [=byte=] whose the [=byte/value=] is the number corresponding to |segment|'s
+    |position|th [=code unit=], according to the [[base64]] encoding.
+
+    NOTE: The two most significant bits of |first| are 0.
+1. Let |sign| be 1 if |first| & 0x01 is 0x00, and -1 otherwise.
+1. Let |value| be (|first| &gt;&gt; 1) & 0x0F, as a number.
+1. Let |nextShift| be 4.
+1. Let |currentByte| be |first|.
+1. While |currentByte| & 0x20 is 0x20:
+    1. Advance |position| by 1.
+    1. If |position| points to the end of |segment|, throw an error.
+    1. Set |currentByte| to the [=byte=] whose the [=byte/value=] is the number corresponding to
+        |segment|'s |position|th [=code unit=], according to the [[base64]] encoding.
+    1. Let |chunk| be |currentByte| & 0x1F, as a number.
+    1. Add |chunk| &lt;&lt; |nextShift| to |value|.
+    1. If |value| is greater than or equal to 2<sup>31</sup>, throw an error.
+    1. Increase |nextShift| by 5.
+1. Advance |position| by 1.
+1. Return |value| * |sign|.
+
+NOTE: In addition to returning the decoded value, this algorithm updates the [=position variable=]
```

Comment:
Can we just return a tuple?

Reply:
This is based on an existing pattern in other bikeshed/web specs to use this "dynamic variable", defined at https://infra.spec.whatwg.org/#string-position-variable